bin/build.ts

```diff
@@ -30,6 +36,68 @@ const IMPORTER = (filePath: string) => {
   return import(filePath)
 }
 
+const ogTemplate = await readFile(joinToURL(import.meta.url, '../assets/og_template.svg'), 'utf-8')
+
+async function generateOgImage(entry: ReturnType<Collection['all']>[0], htmlOutput: string) {
+  /**
+   * Read the HTML file to extract the meta description
+   */
+  const htmlLocation = joinToURL(import.meta.url, '../', 'dist', htmlOutput)
+  const html = await readFile(htmlLocation, 'utf-8')
+  const metaDescription = html.match(/<meta name="og:description" content="(.*)">/)
+  const description = metaDescription ? metaDescription[1] : ''
+
+  /**
+   * Create the og directory when missing
+   */
+  if (!existsSync('public/og')) await mkdir('public/og', { recursive: true })
+
+  /**
+   * Skip generating the og image when it already exists
+   */
+  const slugifiedTitle = string.slug(entry.title, { lower: true })
+  const output = `public/og/${slugifiedTitle}.png`
+  let shouldGenerateOgImage = true
+  if (existsSync(output)) {
+    shouldGenerateOgImage = false
+  }
+
+  /**
+   * Take 3 lines of 40 characters each
+   * insert them in the svg template
+   */
+  const lines = description
+    .trim()
+    .split(/(.{0,60})(?:\s|$)/g)
+    .filter(Boolean)
+
+  const svg = ogTemplate
+    // max 24 characters for the title
+    .replace('{{ title }}', entry.title.slice(0, 24))
+    .replace('{{ line1 }}', lines[0])
+    .replace('{{ line2 }}', lines[1] || '')
+    .replace('{{ line3 }}', lines[2] || '')
+
+  try {
+    if (shouldGenerateOgImage) {
+      await sharp(Buffer.from(svg))
+        .resize(1200 * 1.1, 630 * 1.1)
+        .png()
+        .toFile(output)
+    }
+  } catch (e) {
+    console.error('Failed to generate og image', e)
+  }
+
+  /**
+   * Insert the og:image meta tag in the output
+   */
+  const ogImageUrl = output.replace('public/', 'https://feat-og-images-v2.v6-docs.pages.dev')
```

Comment:
i've updated the base url so we can see what it looks like. we should not forget to change this before merging

Reply:
Does the URL has to be an absolute path with the domain?. Can't we have a path with a slash `output.replace('public/', '/')`

Reply:
It has to be an absolute path for third-party content to read it

Reply:
Okay.